enterprise/internal/campaigns/reconciler.go

```diff
@@ -662,7 +669,8 @@ func determinePlan(ctx context.Context, tx *Store, ch *campaigns.Changeset) (*pl
 			pl.SetOp(operationReopen)
 		}
 
-		if delta.draftChanged {
+		// Only do undraft, when the codehost supports draft changesets.
+		if delta.draftChanged && campaigns.ExternalServiceSupports(ch.ExternalServiceType, campaigns.CodehostCapabilityDraftChangesets) {
```

Comment:
Tiniest nitpick of all, but reading this again now, I think this name might be better:

```suggestion
		if delta.undrafted && campaigns.ExternalServiceSupports(ch.ExternalServiceType, campaigns.CodehostCapabilityDraftChangesets) {
```

Because `draftChanged` can also be understood as going from `publish: false` to `publish: "draft"` and then it's confusing to read `AddOp(operationUndraft)`